scenemanager/src/main/java/io/pixsight/scenemanager/SceneManager.kt

```diff
@@ -412,21 +451,53 @@ object SceneManager {
         }
         val scenesIdsToViews = meta.scenesIdsToViews
 
-        meta.sceneAnimationAdapter
-            .doChangeScene(scenesIdsToViews, meta.scenesParams, sceneId, animate)
-        notifyListener(meta.currentSceneId, sceneId, meta.listener)
-        meta.currentSceneId = sceneId
+        // inflate on demand
+        if (meta.inflateOnDemand) {
+            val currentSceneViews = scenesIdsToViews.get(sceneId)
+            val inflatedViews = ArrayList<Pair<Int, View>>()
+            currentSceneViews?.forEach {
+                if (it is InflateOnDemandLayout) {
+                    val view = it.inflate()!!
+                    // The animation adapter may want to change the default attribute
+                    // to avoid the view from blinking. (before doChangeScene is called)
+                    meta.sceneAnimationAdapter.onViewInflatedOnDemand(sceneId, view)
+                    inflatedViews.add(Pair(it.id, view))
+                }
+            }
+
+            scenesIdsToViews.forEach { _, sceneViews ->
+                inflatedViews.forEach { pair ->
+                    if (sceneViews.removeAll { it.id == pair.first }) {
```

Comment:
Same here, what are pair and it ?

Reply:
Je remplace les InflateOnDemandLayout dans les metas par les vues qui viennent d'etre inflate.
J'ai ajouté un commentaire et renommé les variables.